frontend/components/AddFundsThroughBridge/useGenerateAddFunds.tsx

```diff
@@ -0,0 +1,126 @@
+import { uniq } from 'lodash';
+import { useCallback, useMemo, useState } from 'react';
+
+import { AddressBalanceRecord } from '@/client';
+import { AddressZero } from '@/constants/address';
+import { TokenSymbol } from '@/enums/Token';
+import { useServices } from '@/hooks/useServices';
+import { Address } from '@/types/Address';
+import { BridgeRequest } from '@/types/Bridge';
+import { typedKeys } from '@/types/Util';
+import { getTokenDetailsFromAddress } from '@/utils/middlewareHelpers';
+
+import { DefaultTokenAmount } from './types';
+import { useAddFundsGetBridgeRequirementsParams } from './useAddFundsGetBridgeRequirementsParams';
+
+export type GeneratedInput = {
+  tokenAddress: Address;
+  symbol: TokenSymbol;
+  amount: number;
+};
+
+/**
+ * Generate inputs to add funds for bridging.
+ *
+ * Example: If the chain is Gnosis, user can add OLAS and XDAI.
+ * This function will return an array of objects with tokenAddress, symbol, and amount set to 0.
+ */
+export const useGenerateAddFunds = ({
+  requirements,
+  defaultTokenAmounts,
+  destinationAddress,
+  onlyNativeToken = false,
+}: {
+  requirements: AddressBalanceRecord;
+  defaultTokenAmounts?: DefaultTokenAmount[];
+  destinationAddress?: Address;
+  onlyNativeToken?: boolean;
+}) => {
+  const { selectedAgentConfig } = useServices();
+  const toMiddlewareChain = selectedAgentConfig.middlewareHomeChainId;
+  const getBridgeRequirementsParams =
+    useAddFundsGetBridgeRequirementsParams(destinationAddress);
+
+  // In general, master_safe and master_eoa addresses are used.
```

Comment:
since the requirements are passed as a prop, I don't think we should mention what is expected to be in the prop here, might be confusing